pss/pss.go

```diff
@@ -543,34 +571,28 @@ func (p *Pss) enqueue(msg *PssMsg) error {
 //
 // Will fail if raw messages are disallowed
 func (p *Pss) SendRaw(address PssAddress, topic Topic, msg []byte) error {
+	defer metrics.GetOrRegisterResettingTimer("pss.send.raw", nil).UpdateSince(time.Now())
+
 	if err := validateAddress(address); err != nil {
 		return err
 	}
+
 	pssMsgParams := &msgParams{
 		raw: true,
 	}
 	payload := &whisper.Envelope{
 		Data:  msg,
 		Topic: whisper.TopicType(topic),
 	}
+
 	pssMsg := newPssMsg(pssMsgParams)
 	pssMsg.To = address
 	pssMsg.Expire = uint32(time.Now().Add(p.msgTTL).Unix())
 	pssMsg.Payload = payload
+
 	p.addFwdCache(pssMsg)
-	err := p.enqueue(pssMsg)
-	if err != nil {
-		return err
-	}
 
-	// if we have a proxhandler on this topic
-	// also deliver message to ourselves
-	if capabilities, ok := p.getTopicHandlerCaps(topic); ok {
-		if p.isSelfPossibleRecipient(pssMsg, true) && capabilities.prox {
-			return p.process(pssMsg, true, true)
-		}
-	}
-	return nil
+	return p.enqueue(pssMsg)
```

Comment:
self delivery is indeed a contraversial thing. 
For direct send to address it is indeed not needed but if we get rid of it. then prox short circuit test should be removed and the pushsync pusher should be aware of kademlia depth and replicate prox message pickup logic, which we thought was ugly...

Reply:
Handling messages when a client calls `Send` is absolutely not intuitive, let alone doing it synchronously.

If you want that kind of functionality, I think there should be another client API that makes this more explicit - if some kind of processing is needed, maybe we should call it `pss.Process(msg)` or something else.

I don't know what `prox short circuit test` is so will probably wait for @nolash.

The `Pusher` on the `pushsync-anton` PR is indeed amended to check if it is the closest peer to a chunk. I don't think this responsibility should be delegated to a `Send` method.

Reply:
> The Pusher on the pushsync-anton PR is indeed amended to check if it is the closest peer to a chunk

As far as I recall, the deliver to self should happen if the message is in the sender's _neighborhood_, not only if it's the "closest".

Reply:
`Push sync` is supposed to sync a chunk to the `closest peer` (in terms of `po`), whereas `Pull sync` is supposed to replicate a chunk to the NN, AFAIK.

So in the meaning of the word `sync` in Push Sync, we `needToSync` a chunk if there is a `closer peer` to the chunk. We can change that if necessary.

At the moment, if our node is responsible for a given chunk (if it is the closest node in terms of po), then a chunk is immediately marked as `synced`. This could be a problem if we shut down the laptop, as only with Push Sync we wouldn't have replicated all chunks. We would have to rely on Pull Sync to actually sync the streams within the NN.